### PR TITLE
Move test_dns_multiple_addresses to a new test group

### PR DESCRIPTION
**Pull request by gkwicker**
Description
-----------

Move the multiple-address DNS test to a separate test group.

Checklist:
----------

- [ X ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/secure_sockets/test/iot_test_tcp.c [hunk 1]
```diff
@@ -440,6 +440,11 @@ size_t xHeapA;
 /* Group declaration required by Unity framework. */
 TEST_GROUP( Full_TCP );
 
+/* Extended test group for tests which aren't required for qualification
+ * but demonstrate additional features or preferred behavior.
+ */
+TEST_GROUP( Full_TCP_Extended );
```
Comment (AniruddhaKanhere) [inline libraries/abstractions/secure_sockets/test/iot_test_tcp.c:446]: Should this be added to [aws_test_runner.c](https://github.com/aws/amazon-freertos/blob/master/tests/common/aws_test_runner.c#L82) as well? I am not sure what is required for release and thus I might be wrong.
If so, we need to add #define in [aws_test_runner_config.h](https://github.com/aws/amazon-freertos/blob/master/vendors/pc/boards/windows/aws_tests/config_files/aws_test_runner_config.h#L56) in vendor directory of all respective vendors.

Comment (gkwicker) [reply libraries/abstractions/secure_sockets/test/iot_test_tcp.c:446]: Thanks @AniruddhaKanhere, we will not add the extended tests here.  These are optional tests for now.

File: libraries/abstractions/secure_sockets/test/iot_test_tcp.c [hunk 2]
```diff
@@ -3262,7 +3262,7 @@ TEST( Full_TCP, AFQP_SOCKETS_htons_HappyCase )
 }
 /*-----------------------------------------------------------*/
 
-TEST( Full_TCP_Extended, test_dns_multiple_addresses )
+TEST( Full_TCP_Extended SOCKETS_dns_multiple_addresses )
```
Comment (AniruddhaKanhere) [inline libraries/abstractions/secure_sockets/test/iot_test_tcp.c:3265]: Oh, you missed a comma here :)
Causing tests to fail

Comment (gkwicker) [reply libraries/abstractions/secure_sockets/test/iot_test_tcp.c]: fixed

**Pull request merged**